.github/workflows/preview-env-fork.yml

```diff
@@ -0,0 +1,104 @@
+          name: pr.txt
+      - name: Save PR number
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        id: pr
+        run: |
+          cat pr.txt
+          echo "::set-output name=id::$(<pr.txt)"
+      - name: Check label
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        uses: actions/github-script@v4
+        id: label-checker
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const query = `query($owner:String!, $name:String!, $pr:Int!) {
+              repository(owner:$owner, name:$name){
+                pullRequest(number:$pr) {
+                  labels(first:50) {
+                    nodes{
+                      name
+                    }
+                  }
+                }
+              }
+            }`;
+            const variables = {
+              owner: 'mattermost',
+              name: 'docs',
+              pr: ${{ steps.pr.outputs.id }}
+            };
+            const result = await github.graphql(query, variables);
+            console.log(result.repository.pullRequest.labels.nodes);
+            const labels = result.repository.pullRequest.labels.nodes;
+            for (var index = 0; index < labels.length; ++index) {
+              var label = labels[index];
+              if(label.name == 'preview-environment') {
+                return true;
+              }
+            }
+            return false;
+
+      - uses: actions/checkout@v2
+        if: ${{ github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        with:
+          ref: ${{ github.event.workflow_run.head_sha }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          submodules: true
+          fetch-depth: 0
+
+      - name: Setup Hugo
+        if: ${{ github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.74.2'
+          # extended: true
+
+      - name: Build Previews
+        if: ${{ github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        run: |
+          rm -rf ./dist
+          hugo -F -s site --baseURL http://mattermost-dev-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/ --destination ../dist/html
+
+      - uses: shallwefootball/s3-upload-action@master
+        name: Upload Preview Env
+        if: ${{ github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        with:
+          aws_key_id: ${{ secrets.AWS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws_bucket: ${{ secrets.AWS_BUCKET }}
+          source_dir: ./dist/html
+          destination_dir: ${{ steps.pr.outputs.id }}
+
+      - name: Add comment to PR
+        uses: peter-evans/create-or-update-comment@v1
```

Comment:
Not sure I like this from a sec perspective.

Reply:
we can review the tagged code and evaluate it. I already did a look and did not see anything that is questionable

Reply:
I think it still could be externally retagged without our notice AFAIU. So I find this questionable. Especially if we provide these sensitive parameters.